### ISSUE TITLE
Change how `default` as a CI provider points at actual providers

### ISSUE THREAD

**Issue by CJ-Wright** [conda-forge/conda-smithy#1309]
We might want to have more flexibility than we currently do in how we set CI providers. Currently default is [hard coded](https://github.com/conda-forge/conda-smithy/blob/149a3cd2d54a3174bc43cfb4b4070b7906a88274/conda_smithy/configure_feedstock.py#L1480) to mean `native` for most of the CI providers. We might consider changing that so that default points to whatever we want it to point to more dynamically. 

This would greatly help with the powerpc issues we're running into today, where TravisCI is having issues. Many packages could run on emulated hardware without issue but currently require a manual change in the CI providers to `azure`. It would be good to have this change more centrally, so that if the current provider that is our `default` goes down we can rapidly switch to a different provider.

I propose a hierarchy of providers: 
1. default: this means that the package doesn't care what the provider is (emulated, bare metal, cross compiled) and conda-forge is free to select whatever is best (most available, lowest cost, etc.)
2. emulated, native, cross: for packages that need to declare that they need emulated hardware or actual hardware or a cross compiler but don't care about which provider actually is used. For instance if a new provider came out for power we could move packages that must have an actual PowerPC machine to that provider.
3. The actual provider name: if the feedstock needs a singular provider, for instance it only runs on circleci (or appveyor) the feedstock can declare that it needs that particular provider.

I think the major changes here are:
1. the default hard coding would be removed from `native` and `emulated`
1. the setting of non-provider names would be in a config file somewhere
1. adding the cross compiler option?

Config file example
```yaml
defaults:
  linux_aarch64: emulated
  linux: native
...
emulated: 
  linux_aarch64: azure
  linux: azure
...
native:
  linux_aarch64: drone
  linux_ppc64le: travis
```

I think most of the code is there with just a few changes to how these are handled and configured.

This also helps mitigate the operational risk of a CI going down since we can rapidly change which service is providing which architectures. It also gives us better metadata, since we'll know exactly which feedstocks *need* true hardware and which don't care.

**Comment by isuruf** [comment]
Is that config file supposed to be in a feedstock?

**Comment by CJ-Wright** [comment]
I think it would be best in a central location so that we can change providers across feedstocks. It could still be in smithy if we feel ok cutting smithy releases just to change the config file.

We could also put it in ci provider.

**Comment by isuruf** [comment]
Ah, so the feedstock config file remains the same?

**Comment by CJ-Wright** [comment]
Right, the feedstock says `default` and then we translate that into an actual provider, whatever is most convenient at the time (as declared by this config thing).

**Comment by isuruf** [comment]
Thanks for clarifying. I'm okay with releasing conda-smithy just to make this change.

**Comment by carterbox** [comment]
What is the status on this implementation? I just ran into this issue because I was confused about why https://github.com/conda-forge/libwebp-base-feedstock/pull/8 wasn't shipping powerpc builds when the conda-forge.yaml states `provider: {linux_aarch64: default, linux_ppc64le: default}`. I'm confused because the ARM builds are emulated on azure by default but the powerpc builds aren't?

Can I help?

**Comment by isuruf** [comment]
That's unrelated. ppc64le builds are not running right now because we ran out of credits in Travis.

**Comment by carterbox** [comment]
> ppc64le builds are not running right now because we ran out of credits in Travis.

Thanks for the info, but if this "dynamic defaults" feature was implemented, I expect my build would have been pushed to emulation on Azure and I wouldn't be here. I'm still offering to help.

**Comment by isuruf** [comment]
We can already do that by changing https://github.com/conda-forge/conda-smithy/blob/main/conda_smithy/configure_feedstock.py#L1852-L1862
